### PR TITLE
[RFC] restructure the repository to match the latest ansible collections spec

### PR DESCRIPTION
**Pull request by evgeni**
this also adds ansible galaxy metadata and tests that the collection is buildable using mazer (https://github.com/ansible/mazer)

[ansible collections](https://github.com/ansible/ansible/blob/stable-2.8/changelogs/CHANGELOG-v2.8.rst#major-changes) are not well documented yet (https://github.com/ansible/ansible/issues/54420), but you can see examples at https://github.com/ansible-collections

this PR should "just" work if you only check out this repository. if you use the repository inside another project (as a submodule or something), you'll need to adjust `ansible.cfg` of your project to find the modules and module_utils in their new paths.

### CODE REVIEW

**Comment by mdellweg** [comment]
I don't know the 'the latest ansible collections spec', but it looks sane to me.
ping @Fobhep

**Comment by evgeni** [comment]
@mdellweg yeah, the spec is reverse engineered from the examples, as there is no doc yet :(

**Comment by ehelms** [comment]
This will require any current users to update their ansible.cfg for the new library path?

**Comment by evgeni** [comment]
> This will require any current users to update their ansible.cfg for the new library path?

If they have a custom one (and don't use the one from this repo), yes.

File: .travis.yml [hunk 1]
```diff
@@ -10,4 +10,6 @@ addons:
     packages:
     - rpm
 install: make test-setup
-script: make lint sanity test
+script:
+  - make lint sanity test
+  - mazer build
```
Comment (ehelms) [inline .travis.yml]: Just a thought, add a make build step to keep make as the interface?

**Comment by Fobhep** [comment]
> I don't know the 'the latest ansible collections spec', but it looks sane to me.
> ping @Fobhep

Yeah, lgtm as well!

File: galaxy.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   - "Chris Forkner <gen2fish@gmail.com>"
+   - "Eric D. Helms <ericdhelms@gmail.com>"
+   - "Evgeni Golov <evgeni@golov.de>"
+   - "Greg Swift <gregswift@gmail.com>"
+   - "Manuel Bonk <githubMB@bonkii.com>"
+   - "Markus Bucher <bucher@atix.de>"
+   - "Matthias Dellweg <dellweg@atix.de>"
+   - "Philipp <philipp98.joos@gmail.com>"
+   - "Quirin Pamp <Q.Pamp.2009@my.bristol.ac.uk>"
+   - "Sean O'Keeffe <seanokeeffe797@gmail.com>"
```
Comment (mdellweg) [inline galaxy.yml]: spacing is inconsistent here (2 vs 3 spaces)

File: Makefile [hunk 3]
```diff
@@ -56,10 +56,13 @@ debug-setup: .tmp/ansible
 test-setup: test/test_playbooks/server_vars.yml
 	pip install --upgrade pip
 	pip install -r requirements-dev.txt
-	pip install -r https://raw.githubusercontent.com/ansible/ansible/devel/requirements.txt 
+	pip install -r https://raw.githubusercontent.com/ansible/ansible/devel/requirements.txt
 test/test_playbooks/server_vars.yml:
 	cp test/test_playbooks/server_vars.yml.example test/test_playbooks/server_vars.yml
 
+dist:
```
Comment (mdellweg) [inline Makefile]: Can we have this in `make help`?

**Comment by ehelms** [comment]
I say rebase it and let's merge it so we get used to the new structure. This will cause almost all PRs to need to rebase but thats acceptable.

**Comment by evgeni** [comment]
Now that there are docs (https://galaxy.ansible.com/docs/contributing/creating_collections.html) it's probably time to rebase this PR (there are new modules since I've opened it) and merge it.

Anyone with merge access should be able to push to my fork, so please do. Otherwise I'll come back to this in 2 weeks

File: galaxy.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  - "Quirin Pamp <Q.Pamp.2009@my.bristol.ac.uk>"
+  - "Sean O'Keeffe <seanokeeffe797@gmail.com>"
+version: "0.0.1"
+license: "GPL-3.0-or-later"
+tags:
+  - foreman
+  - katello
+  - satellite
+dependencies:
+  ansible.version: ">= 2.3.0"
```
[the remark this answers is not in the document]
Comment (evgeni) [reply galaxy.yml]: No, as we kept the old aliases

**Comment by ehelms** [comment]
@mdellweg Thanks! You beat me to it, and I appreciate it

**Pull request merged**